internal/internal_event_handlers.go

```diff
@@ -881,6 +886,41 @@ func (wc *workflowEnvironmentImpl) SideEffect(f func() (*commonpb.Payloads, erro
 	wc.logger.Debug("SideEffect Marker added", tagSideEffectID, sideEffectID)
 }
 
+func (wc *workflowEnvironmentImpl) TryUse(flag sdkFlag) bool {
+	return wc.sdkFlags.tryUse(flag, !wc.isReplay)
+}
+
+func (wc *workflowEnvironmentImpl) QueueUpdate(name string, f func()) {
+	wc.blockedUpdates[name] = append(wc.blockedUpdates[name], f)
+}
+
+func (wc *workflowEnvironmentImpl) HandleUpdates(name string) bool {
+	updatesHandled := false
+	if blockedUpdates, ok := wc.blockedUpdates[name]; wc.sdkFlags.tryUse(SDKPriorityUpdateHandling, !wc.isReplay) && ok {
+		updatesHandled = true
+		for _, update := range blockedUpdates {
+			update()
+		}
+		delete(wc.blockedUpdates, name)
+	}
+	return updatesHandled
+}
+
+func (wc *workflowEnvironmentImpl) DrainUnhandledUpdates() bool {
+	rerun := false
+	// Check if any blocked updates remain when we have no more coroutines to run and let them run so they are rejected.
+	// Generally iterating a map in workflow code is bad because it is non deterministic
+	// this case is fine since all these update handles will be rejected and not recorded in history.
```

Comment:
It wasn't immediately clear to me that rejection would happen because without the context of what the functions in `blockedUpdates` really are (wrappers around the update that will also reject if no one registered, not just the user handler) it seems like this might be executing handlers directly.

I'm not sure there's really any great way to deal with that, though.

Reply:
Yeah I can document more  clearly `blockedUpdates` doesn't contain the handlers directly, but spawns a coroutine  to handle them.

>It's also not immediately obvious that it's guaranteed that all updates will be rejected. I guess this is the case because, if the updates were registered, then they would have had their handlers run before all the coroutines block.

Yeah it is up to the handler to reject, I'd rather keep  the rejection logic all in the handler rather then sprinkle it around.

Reply:
>That being the case - perhaps it would make more sense for this function to be called RejectUnhandledUpdates

I didn't name it that because I thought if I added a dynamic update handle then they would be handled here, we don't support dynamic anything in Go yet so maybe that isn't a good reason.